core/src/client/types.rs

```diff
@@ -129,10 +129,12 @@ pub struct Header {
     /// Nonce
     pub nonce: Option<H64>,
     /// Size in bytes
+    #[serde(rename = "sizeHeader")]
```

Comment:
where is this from? 
I understand there is a name collision due to header names, but how so?

Reply:
I had an error when deserializing the response for a `Block`, which is defined like this:

```
pub struct Block {
    #[serde(flatten)]
    pub header: Header,
    pub total_difficulty: U256,
    pub uncles: Vec<H256>,
    pub transactions: BlockTransactions,
    pub size: Option<U256>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub base_fee_per_gas: Option<U256>,
}
```
The errors said there was twice the same name variable for `size` and `base_fee_per_gas`. Renaming them with serde in the Header was the only solution I could find. Is there another way to handle this kind of issues ?

Reply:
How come this does not happen when one requests a block through the RPC?

Reply:
Renaming it here means it is renamed for the whole app, no? which means we break the JSON API spec? Or does it mean we rename the fields if and only if we're deserializing them (meaning when serializing, we don't rename it)?

Reply:
Discussing this with @MartianGreed,`#[serde(flatten)]` inline all variables from the `Block` which results in having `size` and `base_fee_per_gas` twice. Removing flatten solves the issue. Where is flatten needed ?